src/lib/routes/admin-api/public-signup.ts

```diff
@@ -0,0 +1,289 @@
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Public signup tokens'],
+                    operationId: 'createApiToken',
+                    requestBody: createRequestSchema(
+                        'publicSignupTokenCreateSchema',
+                    ),
+                    responses: {
+                        201: createResponseSchema('publicSignupTokenSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: '/tokens/:token/signup',
+            handler: this.addTokenUser,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Public signup tokens'],
+                    operationId: 'addPublicSignupTokenUser',
+                    requestBody: createRequestSchema('createUserSchema'),
+                    responses: {
+                        200: createResponseSchema('userSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'get',
+            path: '/tokens/:token',
+            handler: this.getPublicSignupToken,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Public signup tokens'],
+                    operationId: 'getPublicSignupToken',
+                    responses: {
+                        200: createResponseSchema('publicSignupTokenSchema'),
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'put',
+            path: '/tokens/:token',
+            handler: this.updatePublicSignupToken,
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['Public signup tokens'],
+                    operationId: 'updatePublicSignupToken',
+                    requestBody: createRequestSchema(
+                        'publicSignupTokenUpdateSchema',
+                    ),
+                    responses: {
+                        200: emptyResponse,
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'delete',
+            path: '/tokens/:token',
+            handler: this.deletePublicSignupToken,
+            acceptAnyContentType: true,
+            permission: ADMIN,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['API tokens'],
+                    operationId: 'deletePublicSignupToken',
+                    responses: {
+                        200: emptyResponse,
+                    },
+                }),
+            ],
+        });
+
+        this.route({
+            method: 'post',
+            path: '/tokens/:token/validate',
```

Comment:
Could be `/tokens/validate` and accept a body with the token, but this works too.